src/Microsoft.Tye.Extensions/Dapr/DaprExtension.cs

```diff
@@ -137,6 +137,21 @@ public override Task ProcessAsync(ExtensionContext context, ExtensionConfigurati
                         deployment.Annotations.Add("dapr.io/enabled", "true");
                         deployment.Annotations.Add("dapr.io/id", project.Name);
                         deployment.Annotations.Add("dapr.io/port", (httpBinding.Port ?? 80).ToString(CultureInfo.InvariantCulture));
+
+                        if (!config.Data.Any())
+                        {
+                            continue;
+                        }
+
+                        if (config.Data.TryGetValue("config", out var daprConfig))
+                        {
+                            deployment.Annotations.Add("dapr.io/config", (daprConfig.ToString() ?? "tracing").ToString(CultureInfo.InvariantCulture));
```

Comment:
`daprConfig.ToString() ?? "tracing"` - this doesn't seem like a very good default because it's not something someone could guess. 

What would you think about putting this into the `if` so that we don't add any annotation when this is `null`?

```C#
if (config.Data.TryGetValue("config", out var daprConfig) && daprConfig is object)
{
...
}
```

Same for the block below, I'd rather not put anything in the yaml if it's the default.

Reply:
@rynowak Are you happy if I add as below

```csharp
if (config.Data.TryGetValue("config", out var daprConfig) && daprConfig is object)
{
    deployment.Annotations.Add("dapr.io/config", daprConfig.ToString()?.ToString(CultureInfo.InvariantCulture));
}

if (config.Data.TryGetValue("log-level", out var logLevel) && logLevel is object)
{
    deployment.Annotations.Add("dapr.io/log-level", logLevel.ToString()?.ToString(CultureInfo.InvariantCulture));
}
```

Reply:
Currently, we have checked `is object` in the if statement, but the `daprConfig` and `logLevel` is possible null and now allow to build. Could I put `#pragma warning disable CS8604 // Possible null reference argument.` to disable this check?

Reply:
I think the reason why you're getting that warning is because you're calling `ToString()` twice.

You can simplify `logLevel.ToString()?.ToString(CultureInfo.InvariantCulture)` to `logLevel.ToString()`.

If that doesn't fix it, feel free to suppress the warning because that seems like a bug.

Reply:
After I do a suppress that warning then everything built okay, except for the format tool :|